### ISSUE TITLE
What is the difference between y and y_cls?

### ISSUE THREAD

**Issue by mrlzla** [Lextal/pspnet-pytorch#2]
Hi!
I am trying to understand the train stub code and I am confused a little bit what is the difference between y and y_cls?

**Comment by Lextal** [comment]
I am training the segmentation with "deep supervision" auxiliary loss as in the paper. **y** is the output for segmentation, and **y_cls** is for auxiliary classification.

**Comment by mrlzla** [comment]
That's good!
AFAIK Segmentation mask is used as y. What should we use as y_cls in order to train the model?

**Comment by Lextal** [comment]
y_cls is a BxN LongTensor where B is batch size and N is a total number of classes in the dataset. It makes sense to modify the loader to yield x, y, y_cls tuples.
Added that to README for future reference, thank you.

**Comment by NicWickman** [comment]
I'm not sure I understand. Can you please give an example of how you would create y_cls? I'm starting from the simple CocoDetection loader that Pytorch provides:

```
    loader = torch.utils.data.DataLoader(data, batch_size=1, shuffle=True, num_workers=4)

    for epoch in range(starting_epoch, starting_epoch + epochs):
        seg_criterion = nn.NLLLoss2d(weight=class_weights)
        cls_criterion = nn.BCEWithLogitsLoss(weight=class_weights)
        epoch_losses = []
        net.train()
        for x, y, y_cls in loader[0]:
```
